keras_cv/models/mobilenet_v3.py

```diff
@@ -179,104 +176,86 @@ def InvertedResBlock(
-                    ratio=se_ratio,
-                    squeeze_activation="relu",
-                    excite_activation="hard_sigmoid",
-                )(x)
+    if block_id:
+        prefix = f"expanded_conv_{block_id}"
 
         x = layers.Conv2D(
-            filters,
+            depth(infilters * expansion),
             kernel_size=1,
             padding="same",
             use_bias=False,
-            name=prefix + "project",
+            name=prefix + "expand",
         )(x)
         x = layers.BatchNormalization(
             axis=channel_axis,
             epsilon=1e-3,
             momentum=0.999,
-            name=prefix + "project/BatchNorm",
+            name=prefix + "expand/BatchNorm",
         )(x)
+        x = activation(x)
 
-        if stride == 1 and infilters == filters:
-            x = layers.Add(name=prefix + "Add")([shortcut, x])
+    x = layers.DepthwiseConv2D(
+        kernel_size,
+        strides=stride,
+        padding="same" if stride == 1 else "valid",
+        use_bias=False,
+        name=prefix + "depthwise",
+    )(x)
+    x = layers.BatchNormalization(
+        axis=channel_axis,
+        epsilon=1e-3,
+        momentum=0.999,
+        name=prefix + "depthwise/BatchNorm",
+    )(x)
+    x = activation(x)
 
-        return x
+    if se_ratio:
+        with custom_object_scope({"hard_sigmoid": HardSigmoid}):
+            x = cv_layers.SqueezeAndExcite2D(
+                filters=depth(infilters * expansion),
+                ratio=se_ratio,
+                squeeze_activation="relu",
+                excite_activation="hard_sigmoid",
+            )(x)
 
-    return apply
+    x = layers.Conv2D(
+        filters,
+        kernel_size=1,
+        padding="same",
+        use_bias=False,
+        name=prefix + "project",
+    )(x)
+    x = layers.BatchNormalization(
+        axis=channel_axis,
+        epsilon=1e-3,
+        momentum=0.999,
+        name=prefix + "project/BatchNorm",
+    )(x)
 
+    if stride == 1 and infilters == filters:
+        x = layers.Add(name=prefix + "Add")([shortcut, x])
 
-def MobileNetV3(
-    stack_fn,
-    last_point_ch,
-    include_rescaling,
-    include_top,
-    num_classes=None,
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    alpha=1.0,
-    minimalistic=True,
-    dropout_rate=0.2,
-    classifier_activation="softmax",
-    name="MobileNetV3",
-    **kwargs,
-):
+    return x
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class MobileNetV3(keras.Model):
     """Instantiates the MobileNetV3 architecture.
 
     References:
         - [Searching for MobileNetV3](https://arxiv.org/pdf/1905.02244.pdf) (ICCV 2019)
         - [Based on the Original keras.applications MobileNetv3](https://github.com/keras-team/keras/blob/master/keras/applications/mobilenet_v3.py)
 
-    This function returns a Keras MobileNetV3 model.
+    This class returns a Keras MobileNetV3 model.
```

Comment:
While you're editing this, would you mind also adding type indications to these docstrings?

(See, for example, docstring updates to VGG19 in #1471)